slave_monitor/js/ClusterRunnerSlaveMonitor.js

```diff
@@ -108,17 +108,20 @@ cls._startVisualization = function(visualizers)
 
         var nodes = [], links = [];
         var graphStateChanged = false;
-        visualizers.map(function (visualizer) {
+        visualizers.forEach(function (visualizer) {
             graphStateChanged = visualizer.update() || graphStateChanged;
             nodes = nodes.concat(visualizer.getNodes());
             links = links.concat(visualizer.getLinks());
         });
+        d3.selectAll('.slaveCircle').on('click', function(e, i) {
+            window.open('http://' + e.slaveDatum.url + '/v1', '');
+        });
```

Comment:
I'm assuming you tested this manually and it works, but just want to double check that this won't add a new click handler every time update is called. (i.e., `.on('click', ...` replaces existing click handlers instead of adding new ones.)

I can't remember d3's behavior but I am thinking at least that jQuery requires you to explicitly remove previous click handlers when adding a new one.